### PR TITLE
[Merged by Bors] - Add changelog to intelliJ plugin

### LINKED ISSUE #542: Add changelog to intelliJ extension
Currently, we don't have a changelog in the extension that can be displayed on the website, it would be better if we could just copy the changelog on release from mirrord

### PR DESCRIPTION
**Pull request by infiniteregrets**
- closes #542
- lint changelog and fix `CHANGED` tag to support this
- bump java version

### CODE REVIEW

File: intellij-ext/gradle.properties [hunk 1]
```diff
@@ -8,7 +8,7 @@ pluginVersion = 3.2.0
 
 # See https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 # for insight into build numbers and IntelliJ Platform versions.
-pluginSinceBuild = 213
+pluginSinceBuild = 222
```
Comment (eyalb181) [inline intellij-ext/gradle.properties:11]: Why was this necessary?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,26 +7,43 @@ Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how
 
 ## [Unreleased]
 
+### Added
```
Comment (eyalb181) [inline CHANGELOG.md:10]: Did you use some linter here? Perhaps we should add it to the CI so we don't crash on release?

Comment (infiniteregrets) [reply CHANGELOG.md:10]: I think that should be a separate PR, it will include ignoring tests when *.md files are changed and a linter

File: intellij-ext/gradle.properties (hunk 1 above)
Comment (infiniteregrets) [reply intellij-ext/gradle.properties]: sorry was just testing something with builds, fixed

File: CHANGELOG.md (hunk 2 above)
Comment (eyalb181) [reply CHANGELOG.md:10]: Lets add it to this PR, since if we don't, a typo in the changelog could break our release flow at a very late stage.

Comment (infiniteregrets) [reply CHANGELOG.md:10]: okay sure

File: intellij-ext/gradle.properties [hunk 3]
```diff
@@ -19,7 +19,7 @@ platformVersion = 2022.2
 platformPlugins = org.jetbrains.plugins.go:222.3345.118, org.intellij.intelliLang
 
 # Java language level used to compile sources and to generate the files for - Java 11 is required since 2020.3
-javaVersion = 11
+javaVersion = 17
```
Comment (eyalb181) [inline intellij-ext/gradle.properties:22]: Why is this necessary?

File: intellij-ext/build.gradle.kts [hunk 4]
```diff
@@ -112,6 +124,7 @@ tasks {
     }
 
     publishPlugin {
+        dependsOn("patchChangelog")
```
Comment (eyalb181) [inline intellij-ext/build.gradle.kts:127]: Why is this needed? Don't we update the CHANGELOG manually?

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt [hunk 5]
```diff
@@ -1,19 +1,15 @@
...
-import com.intellij.openapi.actionSystem.ToggleAction
-import com.intellij.openapi.application.ApplicationManager
-import com.intellij.openapi.project.Project
 import com.github.zafarkhaja.semver.Version
 import com.intellij.ide.plugins.PluginManagerConfigurable
 import com.intellij.ide.plugins.PluginManagerCore
 import com.intellij.ide.util.PropertiesComponent
-import com.intellij.notification.NotificationAction
+import com.intellij.notification.*
+import com.intellij.openapi.actionSystem.AnActionEvent
```
Comment (eyalb181) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt:8]: What happened here?

Comment (infiniteregrets) [reply intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt:8]: that was intellij's linter

File: intellij-ext/gradle.properties (hunk 3 above)
Comment (infiniteregrets) [reply intellij-ext/gradle.properties:22]: - The Java configuration specifies sourceCompatibility=11 but IntelliJ Platform 2022.2 requires sourceCompatibility=17.
See: https://jb.gg/intellij-platform-versions

when run `./gradlew buildPlugin`

Comment (infiniteregrets) [reply intellij-ext/gradle.properties:22]: i believe this could be the reason: https://blog.jetbrains.com/platform/2022/08/intellij-project-migrates-to-java-17/
but didn't sound relevant since it says its needed for 2022.3, but here https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html#intellij-platform-based-products-of-recent-ide-versions it says it is needed for 2022.2

File: intellij-ext/build.gradle.kts (hunk 4 above)
Comment (infiniteregrets) [reply intellij-ext/build.gradle.kts:127]: looks like this might not be needed here, I picked this from the template we are using but it works without it as well
